### PR TITLE
More doc fixes for Options<T>.

### PR DESCRIPTION
**Pull request by jviereck**
Replaces a few option within Option<T> and improves docs for transformation.

### CODE REVIEW

File: src/libcore/option.rs [hunk 1]
```diff
@@ -10,9 +10,9 @@
 
 //! Optional values
 //!
-//! Type `Option` represents an optional value: every `Option`
+//! Type `Option<T>` represents an optional value: every `Option<T>`
```
Comment (apasel422) [inline src/libcore/option.rs]: I really don't think these are necessary.

Comment (jviereck) [inline src/libcore/option.rs]: I asked @steveklabnik about this on IRC and he was in favor of adding them everywhere. Quoting from the IRC conversation at https://botbot.me/mozilla/rust/2015-03-27/?msg=35229748&page=24:


> jviereck: The current docs use "option" and "Option" and "Option<T>" when the type is relevant. Maybe all of the options should be renamed to `Option` to make clear the trait is meant and have a consistent wording?
> steveklabnik: it should always be `Option<T>`

Comment (apasel422) [inline src/libcore/option.rs]: Is that going to be the new convention? It seems way too redundant and a bit distracting. Is `HashMap::contains_key`'s documentation going to become

```
Returns true if the `HashMap<K, V, S>` contains a value for the specified key.
```

My preference is to use a lowercase non-Rust-type word when the meaning is obvious, which it is when we're talking about methods on a type, but I guess this needs some kind of guidelines RFC. At the very least, I don't think we need to use the type parameter everywhere.

Comment (jviereck) [inline src/libcore/option.rs]: As I've just entered the Rust community I don't think I have (and don't intend to have) a strong vote here.

@apasel422, @steveklabnik, please let me just know what you both agree on and I am more than happy to adjust the RFC whatever you come up with :)

Comment (steveklabnik) [inline src/libcore/option.rs]: I _thought_ I included this in https://github.com/rust-lang/rfcs/blob/master/text/0505-api-comment-conventions.md, but apparently did not.

We've been preferring the type params always in the single or double param case, and doing it on a case-by-case basis with more than three. `Option<T>` should use it.

**Comment by steveklabnik** [comment]
closing since we did https://github.com/rust-lang/rust/pull/23862

**Comment by jviereck** [comment]
@steveklabnik AFAIKT the changes in #23862 and here are unrelated? Therefore I don't see why this PR is closed with reference to #23862.

Can you please have a second look at it? Thanks! And sorry in case I caused confusion by cross referencing issues though they are not really related content wise :/

**Pull request closed without merging**